pkg/autodiscovery/autoconfig.go

```diff
@@ -102,27 +111,51 @@ func (ac *AutoConfig) StartPolling() {
 	ac.pollerActive = true
 }
 
+// startServiceListening waits on services and templates and process them as they come.
+// It can trigger scheduling decisions or just update its cache.
+func (ac *AutoConfig) startServiceListening() {
+	ac.m.Lock()
+	defer ac.m.Unlock()
+
+	go func() {
+		for {
+			select {
+			case <-ac.listenerStop:
+				ac.healthListening.Deregister()
+				return
+			case <-ac.healthListening.C:
```

Comment:
nits, I like it better when there is a `break`

Reply:
I agree it's more explicit but as it's written this way for every health check I kept it as is for consistency